kin-devplatform-sdk/src/main/java/kin/devplatform/accountmanager/AccountManagerImpl.java

```diff
@@ -245,30 +247,28 @@ public void onError(Exception e) {
 	public void switchAccount(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
 		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks) {
 		Logger.log(new Log().withTag(TAG).put("switchAccount", "start"));
-		RestorableWalletApi restorableWalletApi = new RestorableWalletApi();
 		try {
-			handleRestorable(accountIndex, callback, migrationManagerCallbacks, restorableWalletApi);
+			updateWalletAddressProcess(accountIndex, callback, migrationManagerCallbacks);
 		} catch (ApiException e) {
 			Logger.log(new Log().priority(Log.ERROR).withTag(TAG).text("getIsRestorableWallet: error is " + e));
-			callback.onFailure(ErrorUtil.createWWalletWasNotCreatedInThisAppException());
+			callback.onFailure(ErrorUtil.fromApiException(e));
 		}
 	}
 
 	//TODO maybe need a better method name...
-	private void handleRestorable(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
-		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks,
-		RestorableWalletApi restorableWalletApi) throws ApiException {
-		restorableWalletApi.getIsRestorableWallet(blockchainSource.getPublicAddress(accountIndex),
+	private void updateWalletAddressProcess(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
+		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks) throws ApiException {
+		AuthApi authApi = new AuthApi();
```

Comment:
should expose it via the `AuthDataSource` like in update wallet

Reply:
Will do